serial/components/virt_rx.c

```diff
@@ -157,24 +157,26 @@ void handle_rx() {
         // This is for our RAW mode, char by char
         if (UART_MODE == RAW_MODE) {
             // microkit_dbg_puts("In raw mode virt rx\n");
-            char got_char = chars[0];
+            char got_char[2];
```

Comment:
Doing comparisons on `got_char[0]` everywhere makes it a bit hard to read, instead, I would only define `char atoi_str[2]` when it's actually needed (which is not all cases). Not saying `atoi_str` is the right name but I think you know what I mean.

Reply:
That makes sense, I've added those changes now.